data-tck/src/main/groovy/io/micronaut/data/tck/tests/AbstractCriteriaSpec.groovy

```diff
@@ -201,6 +201,27 @@ abstract class AbstractCriteriaSpec extends Specification {
             "age"    | "countDistinct" | 'COUNT(DISTINCT(test_."age"))'
     }
 
+    /**
+     * Currently ArrayContains criteria is supported only by Azure Cosmos Db.
+     * If we introduce more criteria not supported by default then we can test it here.
+     */
+    void "test unsupported criteria"() {
+        given:
+        PersistentEntityRoot entityRoot = createRoot(criteriaQuery)
+        Specification arrayContainsSpecification = {
+            root, query, cb ->
+                def parameter = cb.parameter(String)
+                ((PersistentEntityCriteriaBuilder)cb).arrayContains(root.get("name"), parameter)
+        } as Specification
+        def arrayContainsPredicate = arrayContainsSpecification.toPredicate(entityRoot, criteriaQuery, criteriaBuilder)
+        when:
+        criteriaQuery.where(arrayContainsPredicate)
+        getSqlQuery(criteriaQuery)
+        then:
+        Throwable ex = thrown()
+        ex.message.contains('ArrayContains is supported only by Azure Cosmos Db')
```

Comment:
I would write something like `not supported by this implementation`.

Reply:
Updated in https://github.com/micronaut-projects/micronaut-data/pull/1813/commits/7b5265feba176abb69f1b4a78160ac4205b6d1be